Source/Scene/QuadtreePrimitive.js

```diff
@@ -435,70 +435,207 @@ define([
+        var northwestChild = tile.northwestChild;
+        var northeastChild = tile.northeastChild;
+        var allAreRenderable = southwestChild.renderable && southeastChild.renderable &&
+                               northwestChild.renderable && northeastChild.renderable;
+        var allAreUpsampled = southwestChild.upsampledFromParent && southeastChild.upsampledFromParent &&
+                              northwestChild.upsampledFromParent && northeastChild.upsampledFromParent;
+
+        if (allAreRenderable) {
+            if (allAreUpsampled) {
+                // No point in rendering the children because they're all upsampled.  Render this tile instead.
                 addTileToRenderList(primitive, tile);
-            } else if (queueChildrenLoadAndDetermineIfChildrenAreAllRenderable(primitive, tile)) {
+
+                if (tile.needsLoading) {
+                    // Rendered tile that's not waiting on children loads with medium priority.
+                    primitive._tileLoadQueueMedium.push(tile);
+                }
+            } else {
                 // SSE is not good enough and children are loaded, so refine.
-                var children = tile.children;
-                // PERFORMANCE_IDEA: traverse children front-to-back so we can avoid sorting by distance later.
-                for (i = 0, len = children.length; i < len; ++i) {
-                    if (tileProvider.computeTileVisibility(children[i], frameState, occluders) !== Visibility.NONE) {
-                        traversalQueue.enqueue(children[i]);
-                    } else {
-                        ++debug.tilesCulled;
-                    }
+                // No need to add the children to the load queue because they'll be added (if necessary) when they're visited.
+                visitVisibleChildrenNearToFar(primitive, southwestChild, southeastChild, northwestChild, northeastChild, frameState);
+
+                if (tile.needsLoading) {
+                    // Tile is not rendered, so load it with low priority.
+                    primitive._tileLoadQueueLow.push(tile);
                 }
+            }
+        } else {
+            // We'd like to refine but can't because not all of our children are renderable.  Load the refinement blockers with high priority and
+            // render this tile in the meantime.
+            queueChildLoadNearToFar(primitive, frameState.camera.positionCartographic, southwestChild, southeastChild, northwestChild, northeastChild);
+            addTileToRenderList(primitive, tile);
+
+            if (tile.needsLoading) {
+                // We will refine this tile when it's possible, so load this tile only with low priority.
+                primitive._tileLoadQueueLow.push(tile);
+            }
+        }
+    }
+
+    function queueChildLoadNearToFar(primitive, cameraPosition, southwest, southeast, northwest, northeast) {
+        if (cameraPosition.longitude < southwest.east) {
```

Comment:
This is fine if you want to keep it like this with explicit SW, SE, NW, and NE properties, but I believe the way the orignial paper implemented this was to have an array of 4 children (could have enums for SW, SE, NW, and NE), and several arrays of indices (in the range 0 to 3) for each sort order.  This function would then just select the right array of indices based on the camera position.

Up to you but worth considering.